iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java

```diff
@@ -121,6 +124,23 @@ public void export(Network network, Properties parameters, DataSource dataSource
         }
     }
 
+    @Override
+    public void export(Network network, Properties parameters, DataStore dataStore, String filename) {
+        if (network == null) {
+            throw new IllegalArgumentException("network is null");
+        }
+
+        ExportOptions options = createExportOptions(parameters);
+        try {
+            long startTime = System.currentTimeMillis();
+            XmlDataResolver resolver = new XmlDataResolver();
+            NetworkXml.write(network, options, dataStore, resolver.checkFileExtension(filename) ? filename : Files.getNameWithoutExtension(filename) + ".xiidm");
```

Comment:
It disturbing to change the filename a user would like to create, because the extension is not the good one